corehq/util/celery_utils.py

```diff
@@ -186,3 +190,41 @@ def get_running_workers(timeout=10):
         worker_names.extend(worker_info.keys())
 
     return worker_names
+
+
+class LoadBasedAutoscaler(Autoscaler):
+    def _maybe_scale(self, req=None):
+        procs = self.processes
+        cur = min(self.qty, self.max_concurrency)
+
+        available_cpus = multiprocessing.cpu_count()
+        load_avgs = os.getloadavg()
+        max_avg = max(load_avgs[0], load_avgs[1])
+        normalized_load = max_avg / available_cpus
+
+        if cur > procs and normalized_load < 0.90:
+            self.scale_up(cur - procs)
+            return True
+        elif cur < procs or normalized_load > 0.90:
```

Comment:
i think we might want to split this into two elif statements, one for cur < procs, and one for normalized_load > 0.90.

otherwise if the load is high but cur > procs we'd end up scaling down a negative number